src/keystore.cpp

```diff
@@ -53,11 +58,11 @@ bool CBasicKeyStore::GetPubKey(const CKeyID &address, CPubKey &vchPubKeyOut) con
     return true;
 }
 
-bool CBasicKeyStore::AddKeyPubKey(const CKey& key, const CPubKey &pubkey)
+bool CBasicKeyStore::AddKeyPubKey(const CKey& key, const CPubKey &pubkey, bool* isColored)
 {
     LOCK(cs_KeyStore);
     mapKeys[pubkey.GetID()] = key;
-    ImplicitlyLearnRelatedKeyScripts(pubkey);
+    ImplicitlyLearnRelatedKeyScripts(pubkey, isColored);
```

Comment:
isColored is not initialised. None of AddKeyPubKey calls have the third parameter, its a pointer, we are accessing it here without checking if it is initialised. it would crash. 
I think we can remove the flag completely and just add two scripts 
```
ImplicitlyLearnRelatedKeyScripts(pubkey);
ImplicitlyLearnRelatedKeyScripts(pubkey, true);

Reply:
Actually we don't need two calls to ImplicitlyLearnRelatedKeyScripts. since OP_COLOR is not allowed inside the redeem script it will be the same for P2SH and CP2SH. we need not change ImplicitlyLearnRelatedKeyScripts or AddKeyPubKey. Sorry, I didn't realise earlier.

Reply:
ok